src/sql/src/normalize.rs

```diff
@@ -559,6 +559,29 @@ macro_rules! generate_extracted_config {
                     Ok(extracted)
                 }
             }
+
+            impl [<$option_ty Extracted>] {
+                #[allow(unused)]
+                fn into_values(self, catalog: &dyn crate::catalog::SessionCatalog) -> Vec<$option_ty<Aug>> {
+                    use [<$option_ty Name>]::*;
+                    let mut options = Vec::new();
+                    $(
+                        let value = self.[<$option_name:snake>];
+                        let values = generate_extracted_config!(
+                            @ifexpr $allow_multiple,
+                            value,
+                            vec![value]
+                        );
+                        for value in values {
+                            if let Some(value) = value.try_into_value(catalog) {
```

Comment:
I don't quite understand what's going on here but: why is it ok to ignore values that fail `try_into_value` and not error or something?

Reply:
I don't blame you, this macro was not easy for me to figure out and update. I'll try and explain it here and in the process maybe I'll come up with a good comment to add that also explains it.

Before my PR this macro allowed you to convert a list of options into some struct that had a field for each option. So for example when processing `CREATE CLUSTER` options, you would start with a `Vec<ClusterOption<Aug>>`.
```Rust
/// An option in a `CREATE CLUSTER` statement.
pub struct ClusterOption<T: AstInfo> {
    pub name: ClusterOptionName,
    pub value: Option<WithOptionValue<T>>,
}
```

You would convert that into a struct that looks like the following (the struct is generated by the macro so there's no where in the code to look for this exact struct):
```Rust
struct ClusterOptionExtracted {
    availability_zones: Option<Vec<String>>,
    disk: Option<bool>,
    managed: Option<bool>,
    ...
}
```

The `Vec` only contains options that were explicitly specified. So a field in `ClusterOptionExtracted` would be `Some(_)` if the option was specified and `None` if it was not.


Now this commit adds the reverse process, it takes a `ClusterOptionExtracted` and converts it back into a `Vec<ClusterOption<Aug>>`. So if the field is `Some(_)` then we need to generate a `ClusterOption<Aug>` and add it to the `Vec`. However, if the field is `None`, then there was no option specified and we have nothing to add back to the list. That's why we ignore the `None` values here, they are not errors they just don't exist as specified options.